orca-retrofit-reactive/src/main/kotlin/com/netflix/spinnaker/orca/reactive/RequestMetrics.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.reactive
+
+import java.time.Duration
+
+/**
+ * Read-only interface allowing access for consumers of request metrics.
+ */
+interface RequestMetrics {
+  val name: String
+  val averageDuration: Duration
+  val errorPercentage: Double
+}
```

Comment:
Read-only portion of the metrics interface. The `BufferStateSupplier` uses this as it doesn't want to update the metrics, only react to them.